### ISSUE TITLE
Support bracket notation on Obj-C collections

### ISSUE THREAD

**Issue by stuartmorgan-g** [dart-lang/native#1444]
It would be very nice, and far more idiomatic, to be able to write `someNSMutableDictionary[key] = value;` the way I can in Obj-C, rather than having to use `someNSMutableDictionarysetObject_forKey_(value, key);` (which is a construction that basically nobody uses in modern Obj-C development).

Similarly for NSMutableArray.

**Comment by liamappelbe** [comment]
Yeah. package:jni has good support for collections. We need the same thing in package:objective_c
